app/models/ability.rb

```diff
@@ -0,0 +1,32 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    return unless user.present?
+
+    can :destroy, Post, author_id: user.id
+    can :destroy, Comment, author_id: user.id
+    return unless user.is?(:admin)
+
+    can :destroy, Post
+    can :destroy, Comment
```

Comment:
- [ ] Great job adding the ability class to make authorization work, unfortunately,  could navigate and use some features of the app, initially, users should be allowed to create posts, comments, and likes of their own without any problem, but this is not the case because I failed to do any of the mentioned. CanCan should be used to prevent users who did not create a post/comment to delete it but every user should see it.

![access](https://github.com/HanaSabih/BlogApp/assets/53124548/d375eba1-1465-4d58-8b64-9a28b02c706c)

![cancan](https://github.com/HanaSabih/BlogApp/assets/53124548/1369223d-5256-47fa-a3c4-1b9219587a22)